### PR TITLE
DTS-485 Add tls support for grpc clients

### PR DESCRIPTION
**Pull request by dj80hd**
Previously, only insecure default configuration  was available for grpc clients.
This PR provides a default configuration for TLS as well.

In addition to the unit tests, this code has been tested with some [e2e tests](https://github.com/spothero/craig/pull/394/files#diff-80e4f1cfacabf5afef6b0280e94c50e993c08949586151655ea619ff0aa5166aR4189-R4190)

### CODE REVIEW

File: sql/postgres_test.go [hunk 1]
```diff
@@ -113,3 +114,9 @@ func TestPostgresConfigBuildConnectionString(t *testing.T) {
 		})
 	}
 }
+
+func TestErrorConnect(t *testing.T) {
```
Comment (dj80hd) [inline sql/postgres_test.go:118]: Added this to make codecov happy

**Review by theisenmark — APPROVED** [review]
LGTM

**Review by greg-mclain — APPROVED** [review]

**Pull request merged**